wgpu-hal/src/dx12/device.rs

```diff
@@ -2237,9 +2237,9 @@ impl crate::Device for super::Device {
         &self,
         fence: &super::Fence,
         value: crate::FenceValue,
-        timeout_ms: u32,
+        timeout: Option<Duration>,
     ) -> Result<bool, crate::DeviceError> {
-        let timeout_duration = Duration::from_millis(timeout_ms as u64);
+        let timeout = timeout.unwrap_or(Duration::MAX);
```

Comment:
If `None` means the same as `Duration::MAX`, why pass duration by `Option` at all?

Reply:
none of the implementations do this right now, but often there's simpler fast/path for no timeout. I think for the low-level hal api this is fine like that, but I _am_ missing docstring on that!